packages/react-core/src/components/Button/__tests__/Button.test.tsx

```diff
@@ -5,145 +5,161 @@ import { render, screen } from '@testing-library/react';
+    expect(screen.getByText(`${variant} Button`)).toHaveClass(`pf-m-${variant}`);
   });
+});
 
-  test('isBlock', () => {
-    const { asFragment } = render(<Button isBlock>Block Button</Button>);
-    expect(asFragment()).toMatchSnapshot();
-  });
+test('Renders with an aria-label with plain button', () => {
+  const label = 'aria-label test';
+  render(<Button aria-label={label} />);
 
-  test('isDisabled', () => {
-    const { asFragment } = render(<Button isDisabled>Disabled Button</Button>);
-    expect(asFragment()).toMatchSnapshot();
-  });
+  expect(screen.getByLabelText(label)).toHaveAccessibleName('aria-label test');
+});
 
-  test('isDanger secondary', () => {
-    const { asFragment } = render(
-      <Button variant="secondary" isDanger>
-        Disabled Button
-      </Button>
-    );
-    expect(asFragment()).toMatchSnapshot();
-  });
+test('Renders with class pf-m-link when variant = link', () => {
+  render(
+    <Button variant={ButtonVariant.link} icon={<CartArrowDownIcon />}>
+      Block Button
+    </Button>
+  );
+  expect(screen.getByText('Block Button')).toHaveClass('pf-m-link');
+});
 
-  test('isDanger link', () => {
-    const { asFragment } = render(
-      <Button variant="link" isDanger>
-        Disabled Button
-      </Button>
-    );
-    expect(asFragment()).toMatchSnapshot();
-  });
+test('Renders with class pf-m-block when isBlock = true', () => {
+  render(<Button isBlock>Block Button</Button>);
+  expect(screen.getByText('Block Button')).toHaveClass('pf-m-block');
+});
 
-  test('isAriaDisabled button', () => {
-    const { asFragment } = render(<Button isAriaDisabled>Disabled yet focusable button</Button>);
-    expect(asFragment()).toMatchSnapshot();
-  });
+test('Renders with class pf-m-disabled when isDisabled = true', () => {
+  render(<Button isDisabled>Disabled Button</Button>);
+  expect(screen.getByText('Disabled Button')).toHaveClass('pf-m-disabled');
+});
 
-  test('isAriaDisabled link button', () => {
-    const { asFragment } = render(
-      <Button isAriaDisabled component="a">
-        Disabled yet focusable button
-      </Button>
-    );
-    expect(asFragment()).toMatchSnapshot();
-  });
+test('Renders with class pf-m-aria-disabled when isAriaDisabled = true', () => {
+  render(<Button isAriaDisabled>Disabled yet focusable button</Button>);
+  expect(screen.getByText('Disabled yet focusable button')).toHaveClass('pf-m-aria-disabled');
+});
 
-  test('isInline', () => {
-    const { asFragment } = render(
-      <Button variant={ButtonVariant.link} isInline>
-        Hovered Button
-      </Button>
-    );
-    expect(asFragment()).toMatchSnapshot();
-  });
+test('Renders with class pf-m-aria-disabled when isAriaDisabled = true and component = a', () => {
+  render(
+    <Button isAriaDisabled component="a">
+      Disabled yet focusable button
+    </Button>
+  );
+  expect(screen.getByText('Disabled yet focusable button')).toHaveClass('pf-m-aria-disabled');
+});
```

Comment:
Since this would already be covered by the previous test, we could update this to test that passing `isDisabled` and `component="a"` doesn't cause the button to be disabled:

```suggestion
test('Does not render with disabled attribute when isDisabled = true and component = a', () => {
  render(
    <Button isDisabled component="a">
      Disabled button
    </Button>
  );
  expect(screen.getByText('Disabled button')).not.toBeDisabled();
});
```

We could also add a couple of tests to check that the `disabled` and `aria-disabled` attributes are being set correctly